pkg/controller/volume/persistentvolume/framework_test.go

```diff
@@ -645,6 +646,7 @@ func newVolume(name, capacity, boundToClaimUID, boundToClaimName string, phase v
 			AccessModes:                   []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce, v1.ReadOnlyMany},
 			PersistentVolumeReclaimPolicy: reclaimPolicy,
 			StorageClassName:              class,
+			VolumeMode:                    &fs,
```

Comment:
why were these changes needed?

Reply:
the existing code can't handle the case where volumeMode is nil && featuregate is enabled, it assumes apiserver (which isn't running in unit tests) will have defaulted it (https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/util.go#L913 and https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/index.go#L274) . This is wrong though, right?...e.g. kubelet could have the feature gate on but apiserver off and then binding/mounting will not work. I am preparing a patch to fix it

Reply:
@liggitt PTAL at latest commit, now even if feature gate is enabled on kubelet/controller-manager, nil volumemode is treated as filesystem.

Reply:
can we revert these changes, then?

Reply:
@liggitt I went back and forth with @msau42 down below in https://github.com/kubernetes/kubernetes/pull/65829#discussion_r233133131, I reverted the changes then un-reverted them again. Reverting these unit test changes means we have to change some logic when volumeMode is nil which she explained we don't really want to do

Reply:
ok